base_station/gui/src/components/Cameras.vue

```diff
@@ -1,140 +1,102 @@
-            return;
-        }
-
-        if(e.keyCode>=49 && e.keyCode<=54)  //keys 1 to 6
-          this.cam_index_1 = e.keyCode-48
-      })
-
-      // Change cam index based on joystick button
-      interval = window.setInterval(() => {
-        const gamepads = navigator.getGamepads()
-        for (let i = 0; i < 2; i++) {
-          const gamepad = gamepads[i]
-          if (gamepad) {
-            if (gamepad.id.includes('Logitech')) {
-              if (gamepad.buttons[JOYSTICK_CONFIG['down_left_button']]['pressed']) {
-                this.cam_index_1 = CAMERA_NUM['down_left_button']
-              } else if (gamepad.buttons[JOYSTICK_CONFIG['up_left_button']]['pressed']) {
-                this.cam_index_1 = CAMERA_NUM['up_left_button']
-              } else if (gamepad.buttons[JOYSTICK_CONFIG['down_middle_button']]['pressed']) {
-                this.cam_index_1 = CAMERA_NUM['down_middle_button']
-              } else if (gamepad.buttons[JOYSTICK_CONFIG['up_middle_button']]['pressed']) {
-                this.cam_index_1 = CAMERA_NUM['up_middle_button']
-              } else if (gamepad.buttons[JOYSTICK_CONFIG['down_right_button']]['pressed']) {
-                this.cam_index_1 = CAMERA_NUM['down_right_button']
-              } else if (gamepad.buttons[JOYSTICK_CONFIG['up_right_button']]['pressed']) {
-                this.cam_index_1 = CAMERA_NUM['up_right_button']
-              }
-            }
-          }
-        }
-        this.$parent.publish('/microscope', {type: "Microscope", streaming: this.microscope_cam === true})
-      }, 250)
     },
 
     methods: {
-      setCamIndex: function (new_index, stream) {
-        if (stream === 1 && (new_index !== this.cam_index_2 || new_index === -1)) {
-          this.cam_index_1 = new_index
-        } 
-        else if (stream === 2 && (new_index !== this.cam_index_1 || new_index === -1)) {
-          this.cam_index_2 = new_index
-        }
+      setCamIndex: function (index) {
+        this.camsEnabled.splice(index, 1, !this.camsEnabled[index])
+        this.sendCameras();
       },
 
-      toggleDualStream: function () {
-        this.dual_stream = !this.dual_stream
-        if (!this.dual_stream) {
-          this.cams[this.cam_index_2] = false
-          this.cam_index_2 = -1
+      sendCameras: function() {
+        let ports = []
+
+        for (let i = 0; i < this.camsEnabled.length; i++) {
+          if (this.camsEnabled[i]) {
+            ports.push(i)
+          }
         }
-      },
 
-      toggleMicroscopeCam: function () {
-        this.microscope_cam = !this.microscope_cam
-      },
+        for (let i = ports.length; i < this.numCams; i++) {
+          ports.push(-1)
+        }
 
-      sendCameras: function() {
-        this.$parent.publish("/cameras_cmd", {
-          'type': 'Cameras',
-          'port_0': this.cam_index_1,
-          'port_1': this.cam_index_2,
+        this.$parent.publish(this.channel, {
+          'type': 'GUICameras',
```

Comment:
does this affect jetson/cameras now?

Reply:
uh kinda? my understanding is that teleop sends GUICameras structs to either /cameras_control_ish or /cameras_control and then from that, they still publish normal Cameras structs to jetson/cameras